woocommerce/google-listings-and-ads
# Attribute Mapping - Sources

### Changes proposed in this Pull Request:

Closes part of #1648 .

This PR implements more sources in the Attributes. Including Metas and product attributes.
Also it implements a cache API system for the Meta Query.
Finally it implements Yoast SEO integration for GTIN and MPN 


### Detailed test instructions:

1. Check all the attributes with `GET wp-json/wc/gla/mc/mapping/sources?attribute={attribute}` and see if if returns the right values. Try to create new metas and attributes to see they appear in the results.
2. Check that after querying an attrubute of type Field like `GET wp-json/wc/gla/mc/mapping/sources?attribute=brand` a new transient is created `_transient_gla_attribute_mapping_meta_fields` with the met values. 
3. Call `DELETE wp-json/wc/gla/mc/mapping/cache` and check transient is deleted. 
4. Query `GET wp-json/wc/gla/mc/mapping/sources?attribute=gtin` and `GET wp-json/wc/gla/mc/mapping/sources?attribute=mpn` See how they don't include Yoast fields.
5. "Install" Yoast SEO for WooCommerce. Since it's a premium plugin you can just add this in your functions.php `define( 'WPSEO_WOO_VERSION', '19.7.2' );`
6. Query again `GET wp-json/wc/gla/mc/mapping/sources?attribute=gtin` and `GET wp-json/wc/gla/mc/mapping/sources?attribute=mpn` See how they now include Yoast fields.
